### DISCUSSION TITLE
Exercise 1: Show that the intersection of C1 and C2 is consistent

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#26] [Exercises]
**Exercise 1:** Let C1 and C2 be two consistent cuts. Show that the intersection of C1 and C2 is consistent

**Comment by susannacifani** [comment]
Let's start by the definition of consistency: a cut $C$ is consistent if and only if: $e \rightarrow e_i \land e_i \in C \Rightarrow e \in C$.
Let’s assume that $C_1$ is consistent, so $e \rightarrow e' \land e' \in C_1 \Rightarrow e \in C_1$.
Let’s assume that $C_2$ is consistent, so $e \rightarrow e' \land e' \in C_2 \Rightarrow e \in C_2$.
Now let’s consider $C_1 \cap C_2$.
Let’s assume that there’s an event $e'$ in $C_1 \cap C_2$, so $e' \in C_1 \cap C_2$.
We saw that $e' \in C_1 \Rightarrow e \in C_1$ and that $e' \in C_2 \Rightarrow e \in C_2$, so since $e' \in C_1 \cap C_2 \Rightarrow e \in C_1 \cap C_2$ with $e \rightarrow e'$.

So since a cut $C$ is consistent if and only if $e \rightarrow e_i \land e_i \in C \Rightarrow e \in C$, we can say that a cut $C_1 \cap C_2$ is consistent if and only if $e \rightarrow e' \land e' \in C_1 \cap C_2 \Rightarrow e \in C_1 \cap C_2$.
That's what we showed.

**Comment by rotiroti** [comment]
We aim to prove the consistency of the intersection, $C_1 \cap C_2$, and this requires us to show that:

$$(e' \in C_1 \cap C_2) \land (e \to e') \Rightarrow e \in C_1 \cap C_2.$$

Let's break down the proof step by step:

- $(e' \in C_1 \cap C_2) \land (e \to e')$, as per the precondition for consistency.
- $(e' \in C_1 \land e' \in C_2)$, by the definition of set intersection.
- $(e \in C_1 \land e \in C_2)$, since $C_1$ and $C_2$ are consistent cuts.
- $(e \in C_1 \cap C_2)$, by the definition of set intersection.

**Comment by rotiroti** [comment]
The previous solutions contain some errors. Following is an updated version of it.

We aim to prove the consistency of the intersection, $C_1 \cap C_2$, and this requires us to show that:

$$(e' \in C_1 \cap C_2) \land (e \to e') \Rightarrow e \in C_1 \cap C_2.$$

Let's break down the proof step by step:

- $(e' \in C_1 \cap C_2) \land (e \to e')$, as per the precondition for consistency.
- $(e' \in C_1 \land e' \in C_2) \land (e \to e')$, by the definition of set intersection.
- $(e' \in C_1 \land e \to e') \land (e' \in C_2 \land e \to e')$, since $C_1$ and $C_2$ are consistent cuts.
- This implies that $(e \in C_1) \land (e \in C_2)$.
- By the definition of set intersection, we can conclude that $e \in C_1 \cap C_2$.

**Comment by cornflexxx** [comment]
Assume $C_1 \cap C_2$ is inconsistent, then there exists 

$$e\in C_1\cap C_2 \land e' \to e \implies e'\not\in C_1 \cap C_2$$

which means that both $C_1$ and $C_2$ contains $e$ but one between $C_1$ and $C_2$  (or both) do not contain $e'$, hence one between $C_1$ and $C_2$ (or both) is inconsistent. $\square$
